bpf/aya-bpf/src/programs/probe.rs

```diff
@@ -3,19 +3,177 @@ use core::ffi::c_void;
 use crate::{bindings::pt_regs, BpfContext};
 
 pub struct ProbeContext {
-    pub regs: *mut pt_regs,
+    pub regs: Regs,
 }
 
 impl ProbeContext {
     pub fn new(ctx: *mut c_void) -> ProbeContext {
         ProbeContext {
-            regs: ctx as *mut pt_regs,
+            regs: Regs::from(ctx)
         }
     }
 }
 
 impl BpfContext for ProbeContext {
     fn as_ptr(&self) -> *mut c_void {
+        self.regs.as_raw_ptr()
+    }
+}
+
+pub struct Regs {
+    regs: *mut pt_regs,
+}
+
+impl From<*mut c_void> for Regs {
+    fn from(ctx: *mut c_void) -> Self {
+        Regs { regs: ctx as *mut pt_regs }
+    }
```

Comment:
This should take `*mut pt_regs` instead, and require casting on the caller

Reply:
I don't expect anyone to use this trait but you are right, I fixed it